File: Java/src/classes/Exercise.java
package classes;

import java.time.LocalDateTime;

public abstract class Exercise {

    protected String name;
    protected int duration;
    protected LocalDateTime date;

    public Exercise(String name, int duration) {
        this.name = name;
        this.duration = duration;
        this.date = LocalDateTime.now();
    }

    public abstract double calcCalories();

    public String showInfo() {
        return name + " - " + duration + " minutes";
    }

    public int getDuration() {
        return duration;
    }

    public String getName() {
        return name;
    }
}

File: Python/classes/exercise.py
from abc import ABC, abstractmethod
from datetime import datetime

class Exercise(ABC):
    
    def __init__(self, name, duration):
        self.name = name           
        self.duration = duration   
        self.date = datetime.now()  
    
    @abstractmethod
    def calc_calories(self):
        pass
    
    def show_info(self):
        return self.name + " - " + str(self.duration) + " minutes"